### PR TITLE
Generation IX Moves

### PR DESCRIPTION
**Pull request by Bassoonian**
Updates the PLA moves to SV standards and adds data for the new generation IX moves.

## Description
Some moves are marked as requiring a new flag that Lunos implemented as part of his gen 9 ability branch. They need to be uncommented to work. A bunch of moves are still missing custom effects because I'm not familiar enough with the battle codebase to do all of them myself and none of them have animations. Double Shock needs the correct string (it has a placeholder one because I couldn't find the one the game used and I didn't have time to test yet).

## **Discord contact info**
Jasper#5206

### CODE REVIEW

File: src/data/text/move_descriptions.h [hunk 1]
```diff
@@ -3042,13 +3042,200 @@ static const u8 sSandsearStormDescription[] = _(
...
+    "A bad joke summons a snowstorm.\n"
+    "The user also switches out.");
+
+static const u8 sTidyUpDescription[] = _(
+    "The user tidies up hazards.\n"
+    "Its Attack and Speed are raised.");
+
+static const u8 sSnowscapeDescription[] = _(
+    "Summons a snowstorm that\n"
+    "strikes every turn.");
```
Comment (AsparagusEduardo) [inline src/data/text/move_descriptions.h:3174]:
```suggestion
    "Summons a snowstorm that\n"
    "lasts for five turns.");
```

File: src/data/text/move_descriptions.h [hunk 2]
```diff
@@ -3042,13 +3042,200 @@ static const u8 sSandsearStormDescription[] = _(
...
+    "The user attacks suddenly,\n"
+    "raising its Speed.");
+
+static const u8 sChillingWaterDescription[] = _(
+    "A shower with ice-cold water\n"
+    "lowers the target's Attack.");
+
+static const u8 sHyperDrillDescription[] = _(
+    "A spinning pointed part pierces\n"
+    "through a foe's Protect.");
```
Comment (AsparagusEduardo) [inline src/data/text/move_descriptions.h:3190]:
```suggestion
    "A spinning pointed part\n"
    "bypasses a foe's Protect.");
```

File: src/data/text/move_descriptions.h [hunk 3]
```diff
@@ -3042,13 +3042,200 @@ static const u8 sSandsearStormDescription[] = _(
...
+
+static const u8 sTwinBeamDescription[] = _(
+    "Mystical beams from the user's\n"
+    "eyes hit twice in a row.");
```
Comment (AsparagusEduardo) [inline src/data/text/move_descriptions.h:3194]:
```suggestion
    "Mystical eye-beams that hit\n"
    "the target twice in a row.");
```

File: src/data/text/move_descriptions.h [hunk 4]
```diff
@@ -3042,13 +3042,200 @@ static const u8 sSandsearStormDescription[] = _(
...
+
+static const u8 sRageFistDescription[] = _(
+    "The more the user has been hit,\n"
+    "the stronger the move.");
```
Comment (AsparagusEduardo) [inline src/data/text/move_descriptions.h:3198]:
```suggestion
    "The more the user has been\n"
    "hit, the stronger the move.");
```

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]
I can't test in-battle effects as I haven't progressed much in the game yet, so I'll leave that part for @ghoulslash to review :eyes:

File: src/data/text/move_descriptions.h [hunk 5]
```diff
@@ -3042,13 +3042,200 @@ static const u8 sSandsearStormDescription[] = _(
...
+    "Sharply boosts offenses and\n"
+    "Speed by using its own HP.");
+
+static const u8 sKowtowCleaveDescription[] = _(
+    "Slashes the foe after kowtowing.\n"
+    "This attack never misses.");
+
+static const u8 sFlowerTrickDescription[] = _(
+    "A bouquet of flowers never misses\n"
+    "and always lands a critical hit.");
```
[the remark this answers is not in the document]
Comment (AsparagusEduardo) [reply src/data/text/move_descriptions.h:3130]: Updated suggestion, as I thought it was a bit confusing.

File: include/config.h [hunk 6]
```diff
@@ -70,6 +70,7 @@
 #define GEN_6 3
 #define GEN_7 4
 #define GEN_8 5
-#define GEN_LATEST GEN_8
+#define GEN_9 6
+#define GEN_LATEST GEN_9
```
[the remark this answers is not in the document]
Comment (AsparagusEduardo) [reply include/config.h:74]: Or at least until we have implemented enough significant changes to justify start having Gen 9 as the new standard.

**Comment by AsparagusEduardo** [comment]
One more thing, remember to update `B_METRONOME_MOVES`.

File: src/battle_tv.c [hunk 7]
```diff
@@ -299,7 +299,9 @@ static const u16 sPoints_MoveEffect[NUM_BATTLE_MOVE_EFFECTS] =
     [EFFECT_WATER_SPORT] = 4,
     [EFFECT_CALM_MIND] = 1,
     [EFFECT_DRAGON_DANCE] = 1,
-    [EFFECT_CAMOUFLAGE] = 3
+    [EFFECT_CAMOUFLAGE] = 3,
+    [EFFECT_SLEEP_HIT] = 1,
+    [EFFECT_SPECIAL_ATTACK_UP_HIT] = 1,
```
[the remark this answers is not in the document]
Comment (Bassoonian) [reply src/battle_tv.c:304]: We don't, but considering that the two new added things are equivalents of mechanics already present in Gen3, I included them out of analogy. I can remove them if you prefer

**Comment by Bassoonian** [comment]
Updated flags and addressed Eduardo's concerns/comments. It's still a bit unclear which moves are forbidden by Metronome, Sleep Talk, Instruct etc. so that data still needs to be added in later once there's more clarity

**Review by ghoulslash — APPROVED** [review]

**Pull request merged**